### ISSUE TITLE
Issue with animated deletion of FieldArray element

### ISSUE THREAD

**Issue by MartinXPN** [react-hook-form/react-hook-form#7733]
### Version Number

7.25.3

### Codesandbox/Expo snack

Can't provide a link as my code is pretty messy, but I'll describe detailed steps to reproduce the issue

### Steps to reproduce

When deleting an element from a `fieldArray` with `remove(index)`, the states diverge between the `fields` and the `watchFieldArray` when the deletion is animated. 

In my case, I was animating the deletion with MUI `TransitionGroup`. The code looked something like this:
 
```ts
  const {control, formState: {errors}, watch} = useFormContext();
  const { fields, append, remove } = useFieldArray({control, name: 'myFields'});
  const watchFieldArray = watch('myFields');
  console.log('fields:', fields, 'watchFieldArray:', watchFieldArray);
  const controlledFields = fields.map((field, index) => {
      return {...field, ...watchFieldArray[index]};
  });

...


    return <>
        <List>
        {/*    <TransitionGroup>*/}
                {controlledFields.map((item, index) => (
                    // <Collapse key={item.id}>
                        <ListItem key={item.id} secondaryAction={<IconButton edge="end" title="Delete" onClick={() => removeLanguage(index)}><CloseIcon /></IconButton>}>
                            <ItemComponent />
                        </ListItem>
                    // </Collapse>
                ))}
        {/*    </TransitionGroup>*/}
            <Button key="add" sx={{textTransform: 'none', marginLeft: 2}} startIcon={<AddIcon/>} onClick={() => addEmptyItem()}>Add</Button>
        </List>
    </>

```

When keeping the `TransitionGroup` and `Collapse`, the fields are removed with animation and during the animation, after calling the `remove(index)` (especially for the last element in the list) the element is rendered while deleting, and the states between `fields` and `watchFieldArray` diverge. The element first gets removed from the `watchFieldArray` and then automatically adds itself to the array again. 

### Expected behaviour

The `fields` and `watchFieldArray` for fieldArrays need to always stay synced even when the addition/removal of elements are animated.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
https://github.com/react-hook-form/react-hook-form/issues/2848

**Comment by Moshyfawn** [comment]
See #7659 & #4558 for some info on the state of RHF integration with animation libs.
TL;DR not in the scope of RHF atm

Also, your issue has been closed due to then lack of a reproduction example. We totally get that you probably have your codebase that's time-consuming to dissect into an isolated codesandbox example, but you can't really expect us to spend our time trying to reproduce your use-case to investigate your issue further.

**Comment by MartinXPN** [comment]
Okay, thanks for the clarification. 
I just wanted to add an issue about react-hook-form + MUI animations not working properly. I guess the library does not handle the `remove` properly with animations and other people have already noticed that (just found from your links - https://github.com/react-hook-form/react-hook-form/issues/6075) so my issue wasn't actually something new.
